### PR TITLE
COMMERCE-9596 Extension point to define a Collection Provider that supplies SKU's as options

### PR DESCRIPTION
**Pull request by dsitu**
https://liferay.atlassian.net/browse/COMMERCE-9596

### CODE REVIEW

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:commerce-functional

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:stop

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Review by luca-pellizzon — COMMENTED** [review]
LGTM
@dsitu can you just put the 2 regen commits togheter pls?
Thanks

Adding for review @andrea-ale-sbarra and @alessiorendina

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by andrea-ale-sbarra** [comment]
@dsitu there's a conflict to solve, I've also asked Brian if we need to sort the fields in the service.xml. Finally it's not clear how this could work with the front store because I don't see any change in the APIs both admin and delivery.

Thank you!

**Comment by dsitu** [comment]
> @dsitu there's a conflict to solve, I've also asked Brian if we need to sort the fields in the service.xml. Finally it's not clear how this could work with the front store because I don't see any change in the APIs both admin and delivery.
> 
> Thank you!

good point, ill add some logic into `CPDefinitionOptionValueRelLocalServiceImpl`.

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by dsitu** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/CPDefinitionIndexer.java [hunk 1]
```diff
@@ -203,9 +203,13 @@ public void postProcessContextBooleanFilter(
 		}
 
 		if (GetterUtil.getBoolean(attributes.get("secure"))) {
-			long commerceChannelId = GetterUtil.getLong(
+			long commerceChannelGroupId = GetterUtil.getLong(
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/CPDefinitionIndexer.java:206]: Just a reminder, we need to update the new indexer as well when it gets merged: https://github.com/liferay-commerce/liferay-portal/pull/4101

**Review by alessiorendina — APPROVED** [review]
Code LGTM, I'm just wondering if it works correctly on the front store.

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:sf

**Comment by dsitu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by dsitu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce

**Comment by luca-pellizzon** [comment]
see https://github.com/liferay-commerce/liferay-portal/pull/4128

**Pull request closed without merging**